src/main/java/UI/Delphi.java

```diff
@@ -0,0 +1,86 @@
+package UI;
+
+import java.util.Scanner;
+
+import Commands.Command;
+import Exceptions.DelphiException;
+import Parser.Parser;
+import Storage.Storage;
+import TaskList.TaskList;
+
+/**
+ * The main class for running the UI.Delphi application.
+ * It manages the core components such as the task list, storage, parser, and user interface.
+ */
+public class Delphi {
+    private final TaskList taskList;
+    private final Storage storage;
+    private final Parser parser;
+    private final Ui ui;
+
+    /**
+     * Constructs a UI.Delphi instance with the specified file path for storage.
+     *
+     * @param filePath The path to the file where task data is stored.
+     */
+    public Delphi(String filePath) {
+        storage = new Storage(filePath);
+        taskList = new TaskList();
+        parser = new Parser();
+        ui = new Ui();
+    }
+
+    /**
+     * Starts the application, handles user input, and manages the task list.
+     * Displays the welcome message, processes commands, and exits upon receiving the exit command.
+     */
+    public void run() {
```

Comment:
used appropriate names for your variables and methods. For example, taskList, storage, and parser are clear noun-based variable names representing things, and method names like run() and getResponse() follow the verb-based naming convention.